text/0009-multilink_command_handling.md

```diff
@@ -0,0 +1,36 @@
+  * Start date: 2018-05-21
+  * Contributors: Achermann Florian, <acfloria@ethz.ch>
+  * Related issues: 
+  
+# Summary
+
+The goal of this document is to present a design on how to handle commands which are sent over different links with significantly different latencies from the ground control station to a vehicle.
+  
+# Motivation
+
+When transmitting commands over multiple links with significantly different latencies to a single vehicle the order in which the commands are sent might not be equal to the order in which they are received. This can lead to an undesired response of the vehicle especially when using commands such as MAV_CMD_DO_CHANGE_ALTITUDE, MAV_CMD_DO_SET_MODE, or MAV_CMD_DO_REPOSITION.
+
+# Detailed Design
+
+A new type of command is created: COMMAND_STAMPED. This command is basically a COMMAND_LONG with additional metadata:
```

Comment:
Why not `COMMAND_INT`? You need a version that can handle higher accuracy coordinates as well as floats.

Reply:
As far as I understood the `COMMAND_INT` is used to represent waypoints and I think for uploading missions this timing is not an issue. I actually don't know how well all the commands can be represented by using a `COMMAND_INT`, but if it can support all different id's then I do not see a reason not use it. If not I could introduce two new commands: `COMMAND_INT_STAMPED` and `COMMAND_LONG_STAMPED`

Reply:
COMMAND_INT is used with anything that needs higher precision, IE `MAV_CMD_DO_REPOSITION` encodes a lat/long coordinate for the vehicle, and is used in some offboard control situations.

Reply:
Ok, then I would add a `COMMAND_INT_STAMPED` and `COMMAND_LONG_STAMPED`